src/main/java/com/conveyal/r5/analyst/TravelTimeReducer.java

```diff
@@ -0,0 +1,141 @@
+     * Definition of non-interpolated percentile: the smallest value in the list such that no more than P percent of
+     * the data is strictly less than the value and at least P percent of the data is less than or equal to that value.
+     * The 100th percentile is defined as the largest value in the list.
+     * See https://en.wikipedia.org/wiki/Percentile#Definitions
+     */
+    // We scale the interval between the beginning and end elements of the array (the min and max values).
+    // In an array with N values this interval is N-1 elements. We should be scaling N-1, which makes the result
+    // always defined even when using a high percentile and low number of elements.  Previously, this caused
+    // an error below when requesting the 95th percentile when times.length = 1 (or any length less than 10).
+    private static int findPercentileIndex(int nElements, double percentile) {
+        // FIXME this should be floor not round. The definition uses ceiling for one-based indexes and we have zero-based indexes.
+        return (int) Math.round(percentile / 100 * nElements);
+    }
+
+    // TODO rename, this does not "record" the travel times, it consumes them or processes them
+    public void recordTravelTimesForTarget (int target, int[] times) {
+        // TODO factor out getPercentiles method for clarity
+        // Sort the times at each target and read off percentiles at the pre-calculated indexes.
+        int[] percentileTravelTimesMinutes = new int[nPercentiles];
+        if (times.length == 1) {
+            // Handle results with no variation
+            // TODO instead of conditionals maybe overload this function to have one version that takes a single int time.
+            Arrays.fill(percentileTravelTimesMinutes, times[0]);
+        } else if (times.length == timesPerDestination) {
+            Arrays.sort(times);
+            for (int p = 0; p < nPercentiles; p++) {
+                int timeSeconds = times[percentileIndexes[p]];
+                if (timeSeconds == FastRaptorWorker.UNREACHED) {
+                    percentileTravelTimesMinutes[p] = FastRaptorWorker.UNREACHED;
+                } else {
+                    // Int divide will floor; this is correct because value 0 has travel times of up to one minute, etc.
+                    // This means that anything less than a cutoff of (say) 60 minutes (in seconds) will have value 59,
+                    // which is what we want. But maybe converting to minutes before we actually export a binary format is tying
+                    // the backend and frontend (which makes use of UInt8 typed arrays) too closely.
+                    int timeMinutes = timeSeconds / 60;
+                    percentileTravelTimesMinutes[p] = timeMinutes;
+                }
+            }
+        } else {
+            throw new ParameterException("Must supply expected number of times or only one time.");
```

Comment:
I might use "travel time values" instead of "times".  As written, it could seem like the error message is missing a word "Must supply X expected number of times..."

Reply:
Should we have a company wide "definitions" document for variable naming and general discussion? We've gotten caught up with this in the past ("grid"s, "scenarios", etc.)

Reply:
Fixed. @trevorgerhardt we should at least add a glossary to the Analysis documentation.